### DISCUSSION TITLE
Lab3 config_read_check/config_write questions

### DISCUSSION THREAD

**Discussion by junshiuanhsieh** [bol-edu/HLS-SOC-Discussions#85] [Q&A]
這邊有幾個疑問

1. tb最後在check ap_done 和 ap_idle 的時候，只要對應的bit是對的就會顯示ok，但如下圖，第二個ok 是在check ap_done，但那時候我也同時把ap_idle拉成 1 了，所以 rdata (6) 才會跟 exp (2) 不一樣，想請問這樣是ok的嗎？

<img width="349" alt="截圖 2023-10-04 19 29 52" src="https://github.com/bol-edu/HLS-SOC-Discussions/assets/58873985/92749c49-c351-425a-b831-3f82a9b7f055">

2. tb 寫完 ap_start之後，awvalid 和 wvalid 就一直維持 1 ，即使我把ap_start 拉回 0，他還是會一直寫 1 進去，想請問這樣是正常的嗎？因為我為了避免ap_start被寫回1 ，只得把wready和awready設成0，但照理講不應該這樣

3. 作業裡面規定只能用一個乘法器 一個加法器，哪請問常數的加減法算在內嗎？

**Comment by boledulab** [comment]
1. 你的設計是對的，testbench 應該要mask uncheck bit.
2. testbench 的做法不對。今天台大的課程我講解 Lab3。
3. 你是指 data_num 的counting （this is incrementor） ？ 除此常數加減法在 Lab3 何處會用到？ Tap 及 data 都是變數

**Reply by JackeyUi** [reply]
我的理解是: tap的adress是從0x20開始存，testbench給的adress要拉到BRAM需要一個-0x20的shift

**Comment by boledulab** [comment]
Yes， the address offset should be done in the FIR design.
